ansibullbot/triagers/ansible.py

```diff
@@ -1447,8 +1448,8 @@ def create_actions(self, iw, actions):
             if comment not in actions.comments:
                 actions.comments.append(comment)
 
-        actions.newlabel = sorted(set([x.encode('ascii') for x in actions.newlabel]))
-        actions.unlabel = sorted(set([x.encode('ascii') for x in actions.unlabel]))
+        actions.newlabel = sorted(set([x.encode('ascii').decode('utf-8') for x in actions.newlabel]))
+        actions.unlabel = sorted(set([x.encode('ascii').decode('utf-8') for x in actions.unlabel]))
```

Comment:
Not sure if this is the only place where this happens but .doing .encode() and .decode() is usually wrong in general purpose code.  For instance, here, it should probably just be 
``` python
        actions.newlabel = sorted(set(actions.newlabel))
        actions.unlabel = sorted(set(actions.unlabel))
```

The times when it does make sense to both decode and encode:
* converting from one encoding to another.  (In this case, you'd start and end with bytes)
* Making sure that a string only contains characters valid in a specific encoding. (In which case you would encode and decode using the same encoding).

Reply:
Yeah, I think it was a validation for ascii in labels. In other places I saw `.encode('ascii', 'ignore')` for dropping non-ASCII  characters.

In general, I agree with you that I need to just drop encode/decode from these lines completely.